src/ReactiveUI.AndroidSupport/ReactiveRecyclerViewAdapter.cs

```diff
@@ -90,7 +90,7 @@ protected override void Dispose(bool disposing)
 
         private TViewModel GetViewModelByPosition(int position)
         {
-            return _list.Count > position ? null : _list.Items.ElementAt(position);
+            return position > _list.Count ? null : _list.Items.ElementAt(position);
```

Comment:
Won't this throw `IndexOutOfBoundsException` here? E.g. in case when `position` equals to `_list.Count`. Probably worth changing `>`  to `>=` and covering this stuff with some unit tests if possible to verify such cases

Reply:
Hard to do unit tests at the moment with Android. Since we are limited to azure DevOps. There is a task for adding them with tutorial how to achieve it.

Reply:
Position is zero based whereas count starts from 1, if it hits null then there is some other problem
The usual code is _list.Items.ElemantAt(position) without the check.  At least it returns an Empty ViewModel for the implementation.  I might have to re-think how this works though.  Hmmmmm  fresh eyes.

ReacvityRecyclerViewAdapter cannot be used directly, it needs to be used in-directly

Thanks for the comment.